tests/ui/trait-bounds/mismatch-fn-trait.stderr

```diff
@@ -68,8 +68,8 @@ LL |     take(f)
    |     required by a bound introduced by this call
    |
    = note: `impl FnOnce(u32)` implements `FnOnce`, but it must implement `FnMut`, which is more general
-   = note: expected a closure with arguments `(u32,)`
-              found a closure with arguments `(i32,)`
+   = note: expected a closure with signature `fn(i32)`
+              found a closure with signature `fn(u32)`
```

Comment:
why does this talk about closures for opaque types?

Reply:
Because whoever implemented the diagnostic note originally hard-coded "closure" here rather than using the sort string of the type or whatever.